src/detect-flowbits.c

```diff
@@ -91,6 +92,86 @@ void DetectFlowbitsRegister (void)
     DetectSetupParseRegexes(PARSE_REGEX, &parse_regex);
 }
 
+static bool DetectFlowbitIsPostmatch(uint8_t cmd)
+{
+    DEBUG_VALIDATE_BUG_ON(cmd >= DETECT_FLOWBITS_CMD_MAX);
+
+    switch (cmd) {
+        case DETECT_FLOWBITS_CMD_TOGGLE:
+        case DETECT_FLOWBITS_CMD_SET:
+        case DETECT_FLOWBITS_CMD_UNSET:
+            return true;
+    }
+    return false;
+}
+
+SCEnumCharMap flowbit_cmds[] = {
+    { "set", DETECT_FLOWBITS_CMD_SET },
+    { "toggle", DETECT_FLOWBITS_CMD_TOGGLE },
+    { "unset", DETECT_FLOWBITS_CMD_UNSET },
+    { "isnotset", DETECT_FLOWBITS_CMD_ISNOTSET },
+    { "isset", DETECT_FLOWBITS_CMD_ISSET },
+};
+
+static inline int DetectFlowbitValidateDo(Signature *s, uint8_t cmd, uint32_t idx)
+{
+    bool postmatch = DetectFlowbitIsPostmatch(cmd);
+    SigMatch *list = postmatch ? s->init_data->smlists[DETECT_SM_LIST_POSTMATCH]
+                               : s->init_data->smlists[DETECT_SM_LIST_MATCH];
+
+    for (SigMatch *sm = list; sm != NULL; sm = sm->next) {
+        if (sm->type != DETECT_FLOWBITS)
+            continue;
+
+        DetectFlowbitsData *fd = (DetectFlowbitsData *)sm->ctx;
+        if ((fd->idx == idx) && (fd->cmd == cmd))
+            return -1;
+    }
+    return 0;
+}
+
+static int DetectFlowbitValidate(Signature *s, DetectFlowbitsData *fd)
+{
+    struct DetectFlowbitInvalidCmdMap_ {
+        uint8_t cmd1;
+        uint8_t cmd2;
+    };
+
+    struct DetectFlowbitInvalidCmdMap_ icmds_map[] = {
+        /* POSTMATCH, MATCH combinations */
+        { DETECT_FLOWBITS_CMD_UNSET, DETECT_FLOWBITS_CMD_ISNOTSET },
+        { DETECT_FLOWBITS_CMD_SET, DETECT_FLOWBITS_CMD_ISSET },
+        /* POSTMATCH, POSTMATCH combinations */
+        { DETECT_FLOWBITS_CMD_SET, DETECT_FLOWBITS_CMD_TOGGLE },
+        { DETECT_FLOWBITS_CMD_UNSET, DETECT_FLOWBITS_CMD_TOGGLE },
+        { DETECT_FLOWBITS_CMD_SET, DETECT_FLOWBITS_CMD_UNSET },
+        /* MATCH, MATCH combinations */
+        { DETECT_FLOWBITS_CMD_ISSET, DETECT_FLOWBITS_CMD_ISNOTSET },
+    };
+
+    for (uint8_t i = 0; i < ARRAY_SIZE(icmds_map); i++) {
```

Comment:
nit : maybe we could avoid the loop by having a switch on `fd->cmd`

and for the case `DETECT_FLOWBITS_CMD_UNSET` call `DetectFlowbitValidateDo` for `DETECT_FLOWBITS_CMD_ISNOTSET` and `DETECT_FLOWBITS_CMD_TOGGLE`